controllers/rabbitmqcluster_controller_test.go

```diff
@@ -1096,6 +1065,104 @@ var _ = Describe("RabbitmqClusterController", func() {
 		})
 	})
 
+	Context("Pause reconciliation", func() {
+		BeforeEach(func() {
+			cluster = &rabbitmqv1beta1.RabbitmqCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "rabbitmq-pause-reconcile",
+					Namespace: defaultNamespace,
+				},
+			}
+			Expect(client.Create(ctx, cluster)).To(Succeed())
+			waitForClusterCreation(ctx, cluster, client)
+		})
+
+		AfterEach(func() {
+			Expect(client.Delete(ctx, cluster)).To(Succeed())
+		})
+
+		It("works", func() {
+			By("skipping reconciling if label is set to true", func() {
+				Expect(updateWithRetry(cluster, func(r *rabbitmqv1beta1.RabbitmqCluster) {
+					r.Labels = map[string]string{"rabbitmq.com/pauseReconciliation": "true"}
+					r.Spec.Service.Type = "LoadBalancer"
+					r.Spec.Rabbitmq.AdditionalConfig = "test=test"
+				})).To(Succeed())
+
+				// service type is unchanged
+				Consistently(func() corev1.ServiceType {
+					svc := service(ctx, cluster, "")
+					return svc.Spec.Type
+				}, 10*time.Second).Should(Equal(corev1.ServiceTypeClusterIP))
+
+				// configMap and statefulSet do not have their update and restart annotations set
+				Consistently(func() map[string]string {
+					cfm := configMap(ctx, cluster, "server-conf")
+					return cfm.Annotations
+				}, 10*time.Second).ShouldNot(HaveKey("rabbitmq.com/serverConfUpdatedAt"))
+				Consistently(func() map[string]string {
+					sts := statefulSet(ctx, cluster)
+					return sts.Annotations
+				}, 10*time.Second).ShouldNot(HaveKey("rabbitmq.com/lastRestartAt"))
```

Comment:
Perhaps an optimisation here could be to expect consistently only in the first expectation. The reasoning here being that, if the first expectation fails after 10 seconds, it's unlikely other expectations are met (in fact test exits right away); on the hand, if the first expectation succeeds, that likely means that the custom resource has been reconciled and subsequent expectations will be met, given that all the fields we are asserting on are set in a single reconcile loop.

Unless you have evidence that this approach causes flakes, I would be in favour of this optimisation, which could save ~20 seconds of test run time.

Reply:
Good point. Updated!